runners/spark/src/main/java/org/apache/beam/runners/spark/translation/EvaluationContext.java

```diff
@@ -139,11 +139,24 @@ public void setCurrentTransform(AppliedPTransform<?, ?, ?> transform) {
         .collect(Collectors.toMap(e -> e.getKey(), e -> ((PCollection) e.getValue()).getCoder()));
   }
 
-  private boolean shouldCache(PValue pvalue) {
-    if ((pvalue instanceof PCollection)
-        && cacheCandidates.containsKey(pvalue)
-        && cacheCandidates.get(pvalue) > 1) {
-      return true;
+  /**
+   * Cache PCollection if {@link #isCacheDisabled()} flag is false and PCollection is used more then
+   * once in Pipeline or forceCache flag is true.
+   *
+   * @param pvalue
+   * @param forceCache if PCollection is not used more than once
+   * @return if PCollection will be cached
+   */
+  public boolean shouldCache(PValue pvalue, boolean forceCache) {
```

Comment:
Can you reason about expected behavior when forceCache is true and the cache is disabled? In which cases we set forceCache to true?

Reply:
it's a user options, if for some reason they want to. @jbonofre might know more, IIRC he brought it up after some folks asked for it in a couple of meetups/conferences.

Reply:
I don't remember exactly the rational, but AFAIR, users wanted to "force" the caching of RDD (disk storage) even if it's only read once.

Reply:
yeah, I remember the same, but why 🤷‍♂️